src/catalua.cpp

```diff
@@ -824,8 +824,86 @@ int call_lua(std::string tocall)
     return err;
 }
 
-void lua_callback(const char *callback_name)
+void lua_delete_global( const char* name ) {
+    lua_State *L = lua_state;
+    lua_pushnil( L );
+    lua_setglobal( L, name );
+}
+
+template<typename ArgType> void lua_callback_store_arg(
+    const int callback_arg_idx, ArgType callback_arg )
+{
+    const char *callback_arg_name = std::string( "callback_arg" + std::to_string(
+                                        callback_arg_idx ) ).c_str();
```

Comment:
You should probably avoid all the raw C-strings and use `std::string` (and for parameters `const std::string &`) instead.

The usage of `callback_arg_name` below invokes undefined behaviour as it is no longer a valid pointer. `c_str` here is invoked on a temporary object and the result is only valid as long as that `std::string` object stays unchanged, but it's deconstructed at the end of this expression.

Reply:
There were some issues with ellipsis ... and non `c_str`, so I used it everywhere. Anyway I decided to refactor the way attributes are transfered to callbacks.